app/main/session-io.js

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2016 Mozilla
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+*/
+
+import fs from 'fs';
+import path from 'path';
+
+import { ipcMain } from 'electron';
+import { get, set, del } from 'object-path';
+import queue from 'queue';
+import colors from 'colors/safe';
+
+import { parseArgs } from '../shared/environment';
+import { logger } from '../shared/logging';
+
+const PROFILE = parseArgs().profile;
+const SESSION_FILE = path.join(PROFILE, 'session.json');
+logger.info(colors.blue('Using'), colors.green(SESSION_FILE));
+
+// Avoid synchronous (potentially frequent) read/writes to/from the filesystem
```

Comment:
I'm not sure. It should definitely not be debouned though, since debouncing can potentially lead to indefinite wait times; I think you mean "throttle" instead. The frontend already throttles sending the app state to the main process for storage, so constantly writing to disk isn't a problem right now. Not sure whether or not it's useful to throttle on both ends at this point.